src/hotspot/share/gc/shared/collectedHeap.hpp

```diff
@@ -194,6 +194,7 @@ class CollectedHeap : public CHeapObj<mtGC> {
     Z,
     Shenandoah
   };
+  constexpr static uint WorkerThreadLimit = (max_jint - 1) / wordSize;
```

Comment:
Having this in CollectedHeap makes it awkward to figure out include dependencies through the cross-macros. I propose that you take the easy way out and just inline the value in the two places for now.

Ultimately, we should have a good place and way to specify constants in the flags system, but I propose that this is done as a separate RFE when more stakeholders have time to think about this.